app/App.js

```diff
@@ -122,12 +123,17 @@ class App extends React.Component<Props, State> {
     });
   };
 
-  healthCheckFlow = () => {
-    const healthCheckIntervalTime = 60000;
-    store.dispatch(checkNodeConnection());
-    this.healthCheckInterval = setInterval(() => {
-      store.dispatch(checkNodeConnection());
-    }, healthCheckIntervalTime);
+  connectionAndMiningFlow = async () => {
+    const connectionAndMiningIntervalTime = 360000;
+    const isConnected = await store.dispatch(checkNodeConnection());
+    isConnected && (await store.dispatch(getMiningStatus()));
+    this.connectionAndMiningInterval = setInterval(async () => {
+      const isConnected = await store.dispatch(checkNodeConnection());
+      const miningStatus = isConnected ? await store.dispatch(getMiningStatus()) : nodeConsts.NOT_MINING;
```

Comment:
no need to ask for mining status in loop, there is logic that asks for mining status only after mining was initiated